src/transactions/OperationFrame.cpp

```diff
@@ -171,7 +171,25 @@ OperationFrame::checkValid(SignatureChecker& signatureChecker, Application& app,
         }
     }
 
-    if (!checkSignature(signatureChecker))
+    bool shouldSkipSigCheck = false;
+
+    if (mOperation.body.type() == CHANGE_TRUST)
+    {
+        for (auto const& opFrame : mParentTx.getOperations())
+        {
+            auto const& body = opFrame->getOperation().body;
+
+            if (body.type() == CREATE_ACCOUNT &&
+                body.createAccountOp().destination == *mOperation.sourceAccount)
+            {
+                shouldSkipSigCheck = true;
+
+                break;
+            }
+        }
+    }
+
+    if (!checkSignature(signatureChecker) && !shouldSkipSigCheck)
```

Comment:
micro-optimization - changing the check order will cause skipping the `checkSignature` call completely in cases of `"crust"`

Reply:
The `checkSignature()` method has a side-effect.  If it's not called, transactions fail to pass the validity check.